### PR TITLE
SONARJAVA-1040 Improve Rule: S864 Report issues only once per statement

### PR DESCRIPTION
**Pull request by merciesa**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

**Comment by m-g-sonar** [comment]
Due to a limitation of the platform and as discussed, it would be better to raise issue on the operator, but at maximum once by line.

File: java-checks/src/main/java/org/sonar/java/checks/OperatorPrecedenceCheck.java [hunk 1]
```diff
@@ -136,7 +149,7 @@ public void visitBinaryExpression(BinaryExpressionTree tree) {
     Tree.Kind peek = stack.peek();
     Tree.Kind kind = getKind(tree);
     if (requiresParenthesis(peek, kind)) {
-      raiseIssue(tree);
+      raiseIssue();
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/OperatorPrecedenceCheck.java]: Raise the issue on the line of the operator, which can be accessed directly using `tree.operatorToken().line()`

**Comment by merciesa** [comment]
We raise the issue per expression, if the line of the operator has not been already reported.

**Comment by m-g-sonar** [comment]
LGTM!

**Pull request closed without merging**